src/context/websocketContext.jsx

```diff
@@ -54,20 +55,62 @@ export const SocketProvider = (props) => {
 
   PubSub.setOnSubscriptionsChange((newTopics) => updateTopicsDebounce(newTopics))
 
-  const onMessage = (message) => {
-    const data = JSON.parse(message.data)
-    if (data.topic === 'heartbeat') return
+  const [overloaded, setOverloaded] = useState(false)
+  const [toastShown, setToastShown] = useState(false)
 
-    if (data.topic === 'server.restart_requested') setServerRestartingVisible(true)
+  // when overloaded is true, activate toast
+  useEffect(() => {
+    if (overloaded)
+      if (!toastShown) {
+        toast.warning(<RefreshToast />, {
+          autoClose: false,
+          closeButton: false,
+        })
+        setToastShown(true)
+      }
 
-    if (data.sender === window.senderId) {
-      return // my own message. ignore
+    return () => {
+      setOverloaded(false)
     }
-    if (data.topic === 'shout' && data?.summary?.text) toast.info(data.summary.text)
+  }, [overloaded, setOverloaded])
 
-    console.log('Event RX', data)
-    PubSub.publish(data.topic, data)
-  }
+  // onMessage is a function that is called when a message comes in from the websocket
+  // it is a closure that keeps track of the number of calls and the last call time
+  let onMessage = (() => {
+    let callCount = 0
+    let lastCall = Date.now()
+
+    return (message) => {
+      // If the function is called more than 10 times per second, return early.
+      if (callCount > 10) {
+        setOverloaded(true)
+        return console.log('WS OVERLOAD!!!!')
+      }
+
+      const data = JSON.parse(message.data)
+      if (data.topic === 'heartbeat') return
+
+      if (data.topic === 'server.restart_requested') setServerRestartingVisible(true)
+
+      if (data.sender === window.senderId) {
+        return // my own message. ignore
+      }
+
+      const now = Date.now()
+      if (now - lastCall < 1000) {
+        callCount += 1
+      } else {
+        callCount = 0
+      }
+
+      lastCall = now
+
+      if (data.topic === 'shout' && data?.summary?.text) toast.info(data.summary.text)
+
+      console.log('Event RX', data)
```

Comment:
Should console.log stay here ?

Reply:
I use this Event RX everytime i debug events :)

Reply:
Historically yes, makes it a lot easier for us to debug. Maybe at some point we can remove it. Probably a bigger discussion if any logs should be visible when built.